test/Ocelot.UnitTests/Kubernetes/KubernetesProviderFactoryTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿using KubeClient;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Ocelot.Configuration;
+using Ocelot.Configuration.Builder;
+using Ocelot.DependencyInjection;
+using Ocelot.Provider.Kubernetes;
+using Ocelot.Provider.Kubernetes.Interfaces;
+using Ocelot.ServiceDiscovery;
+
+namespace Ocelot.UnitTests.Kubernetes;
+
+public class KubernetesProviderFactoryTests : UnitTest
+{
+    private readonly IOcelotBuilder _builder;
+
+    public KubernetesProviderFactoryTests()
+    {
+        _builder = new ServiceCollection().AddOcelot();
+    }
+
+    [Theory]
+    [Trait("Bug", "977")]
+    [InlineData(nameof(PollKube))]
+    [InlineData(nameof(Kube))]
```

Comment:
I suggest splitting it into two `Fact`s because long polling requires separate testing.